modules/003-deckhouse-config/hooks/update_status.go

```diff
@@ -0,0 +1,189 @@
+and get access to each module state.
+
+ModuleConfig status consists of:
+- 'enabled' field - describes a module's enabled state:
+    * Enabled
+    * Disabled
+    * Disabled by script - module is disabled by 'enabled' script
+    * Enabled/Disabled by config - module state is determined by ModuleConfig
+- 'status' field - describes state of the module:
+    * Unknown module name - ModuleConfig resource name is not a known module name.
+    * Running - ModuleRun task is in progress: module starts or reloads.
+    * Ready - helm install for module was successful.
+    * HookError: ... - problem with module's hook.
+    * ModuleError: ... - problem during installing helm chart.
+*/
+
+var _ = sdk.RegisterFunc(&go_hook.HookConfig{
+	Queue: "/modules/deckhouse-config/status",
+	Kubernetes: []go_hook.KubernetesConfig{
+		{
+			Name:                         "configs",
+			ApiVersion:                   "deckhouse.io/v1",
+			Kind:                         "ModuleConfig",
+			FilterFunc:                   filterModuleConfigForStatus,
+			ExecuteHookOnSynchronization: pointer.BoolPtr(true),
+			ExecuteHookOnEvents:          pointer.BoolPtr(false),
+		},
+	},
+	Schedule: []go_hook.ScheduleConfig{
+		{
+			Name:    "update_statuses",
+			Crontab: "*/15 * * * * *",
+		},
+	},
+	Settings: &go_hook.HookConfigSettings{
+		EnableSchedulesOnStartup: true,
+	},
+}, updateModuleConfigStatuses)
+
+// filterModuleConfigForStatus returns name, enabled flag and the current status from ModuleConfig object.
+func filterModuleConfigForStatus(unstructured *unstructured.Unstructured) (go_hook.FilterResult, error) {
+	var cfg d8config_v1.ModuleConfig
+
+	err := sdk.FromUnstructured(unstructured, &cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	// Extract name, spec and status.
+	return &d8config_v1.ModuleConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: cfg.Name,
+		},
+		Spec: d8config_v1.ModuleConfigSpec{
+			Version:  cfg.Spec.Version,
+			Enabled:  cfg.Spec.Enabled,
+			Settings: cfg.Spec.Settings,
+		},
+		Status: cfg.Status,
+	}, nil
+}
+
+const (
+	d8ConfigGroup      = "deckhouse_config_metrics"
+	d8ConfigMetricName = "module_config_obsolete_version"
+)
+
+func updateModuleConfigStatuses(input *go_hook.HookInput) error {
+	allConfigs := snapshotToModuleConfigList(input.Snapshots["configs"])
+
+	// Get bundle name from 'deckhouse' config.
+	bundleName := ""
+	for _, cfg := range allConfigs {
+		if cfg.GetName() == "deckhouse" {
+			if bundle, ok := cfg.Spec.Settings["bundle"].(string); ok {
+				bundleName = strings.Title(bundle)
+			}
+			break
+		}
+	}
+
+	for _, cfg := range allConfigs {
+		moduleStatus := d8config.Service().StatusReporter().ForConfig(cfg, bundleName)
+		statusPatch := makeStatusPatch(cfg, moduleStatus)
+		if statusPatch != nil {
+			input.LogEntry.Infof("Patch /status for %s: change (state=%s, status=%s) to state=%s, status=%s", cfg.GetName(), cfg.Status.State, cfg.Status.Status, statusPatch.State, statusPatch.Status)
```

Comment:
Nit: `Status.Status`? Looks like it is `Status.Message`, we print human-readable precaustions and recommendations there.